app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt

```diff
@@ -1,46 +1,75 @@
 package com.example.superimagesviewer
 
 import android.app.Application
-import android.content.res.AssetManager
-import androidx.lifecycle.MutableLiveData
 import android.graphics.drawable.Drawable
 import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import coil.Coil
+import coil.request.GetRequest
+import com.facebook.*
 import kotlinx.coroutines.*
+import org.json.JSONArray
+import org.json.JSONObject
 
 class MosaicRepository(application: Application) {
 
     private val drawables = mutableListOf<Drawable>()
     private val mosaicsList = MutableLiveData<List<Drawable>>()
+    fun getMosaicsList(): LiveData<List<Drawable>> {
+        return mosaicsList
+    }
 
     init {
-        CoroutineScope(Dispatchers.Main).launch {
-            loadImagesFromAssets(application)
-        }
+        loadImagesFromInstagram(application)
     }
 
-    fun getMosaicsList(): LiveData<List<Drawable>> {
-        return mosaicsList
+    private fun loadImagesFromInstagram(application: Application) {
+        val token: AccessToken = AccessToken.getCurrentAccessToken()
+        GraphRequest(
+            token,
+            "/$INSTAGRAM_ID/media",
+            null,
+            HttpMethod.GET
+        ) { response ->
```

Comment:
Please put this callback into a new method

Reply:
done